components/admin/QuestDetails.tsx

```diff
@@ -0,0 +1,214 @@
+"use client";
+
+import React, {
+  ReactNode,
+  useEffect,
+  useState,
+  FunctionComponent,
+} from "react";
+import styles from "@styles/quests.module.css";
+import { useAccount } from "@starknet-react/core";
+import { hexToDecimal } from "@utils/feltService";
+import { NFTItem, QuestDocument, UserTask } from "types/backTypes";
+import { Skeleton } from "@mui/material";
+import TasksSkeleton from "@components/skeletons/tasksSkeleton";
+import { generateCodeChallenge } from "@utils/codeChallenge";
+import Timer from "@components/quests/timer";
+import NftImage from "@components/quests/nftImage";
+import Task from "@components/quests/task";
+import Reward from "@components/quests/reward";
+import { AdminService } from "@services/authService";
+import { useNotification } from "@context/NotificationProvider";
+
+type QuestDetailsProps = {
+  quest: QuestDocument;
+  taskId?: string;
+  res?: string;
+  errorMsg?: string;
+  setShowDomainPopup: (show: boolean) => void;
+  hasRootDomain: boolean;
+  hasNftReward?: boolean;
+  rewardButtonTitle?: string;
+  onRewardButtonClick?: () => void;
+  overrideDisabledState?: boolean;
+};
+
+const AdminQuestDetails: FunctionComponent<QuestDetailsProps> = ({
+  quest,
+  setShowDomainPopup,
+  hasRootDomain,
+  hasNftReward,
+  rewardButtonTitle,
+  onRewardButtonClick,
+}) => {
+  const { address } = useAccount();
+  const [tasks, setTasks] = useState<UserTask[]>([]);
+  const [showQuiz, setShowQuiz] = useState<ReactNode>();
+  const [customError, setCustomError] = useState<string>("");
+  const { showNotification } = useNotification();
+  const questId = quest?.id?.toString();
+
+  // this fetches all tasks of this quest from db
+  useEffect(() => {
+    if (!questId) return;
+
+    AdminService.getTasksByQuestId(parseInt(questId)).then((data) => {
+      if ((data as UserTask[]).length) setTasks(data as UserTask[]);
+    });
+  }, [questId]);
+
+  const generateOAuthUrl = (task: UserTask): string => {
```

Comment:
Will we use that in another file ? If so I think you can make it a hook in a service and test it.